Feature/m0092 meja UI 2336 image and font cache (#651)
MicroEJ/docs: ApplicationDeveloperGuide/UI/MicroUI/fonts.rst

To help developers identify quickly why a string is rendered with unexpected characters, it is recommended that the font maker sets a default character that is easy to recognize (a symbol, for example, a rectangle).
This character must have the first character index (index ``0`` is allowed).

Caching Generated Fonts
-----------------------

> Fonts converted using the font generator tool can be cached so that they are not rebuilt every time the application is launched.

Author: ```suggestion
Fonts converted using the Font Generator tool can be cached so that they are not rebuilt every time the application is launched.
```

> Fonts converted using the font generator tool can be cached so that they are not rebuilt every time the application is launched.

Author: **F**ont **G**enerator (several occurrences)

> Doing so can significantly speed up the application build phase.

Author: suggestion: This considerably speeds up the application build phase.

> The cache may be enabled or disabled through an option in the run configuration.

Author: actually this option is not available yet. @seon Should we keep this option?

Reviewer: Maybe highlight that by default the cache is enabled but that it can be disabled through and option.

Author: the option is `-Dej.microui.fontConverter.disableCache=true` (in JRE tab)

> TODO How to enable/disable the cache

Author: link to dev guide ?

The font generator tool obeys several rules when choosing whether a font should be converted.

> - If the cache is disabled, all fonts are generated every time the application is launched.

Author: if the option is not selected, remove this line

> - All fonts will be regenerated if the application is launched using another VEE port and the new VEE port uses a different font generator tool.

Author: ```suggestion
- All fonts will be regenerated if the application is launched using another VEE port and the new VEE port uses a different Font Generator.
```

Author: (several occurrences)

> - All fonts will be regenerated if the application is launched using another VEE port and the new VEE port uses a different font generator tool.

Author: remove "tool" and add upper case

- If the generated font does not exist, it will be generated.
- If the source font has been modified since the last time it was converted, the font will be regenerated.
- The font will be regenerated if the destination format or the range has been modified in the `fonts.list` file.

> Cached fonts are stored in ``TODO PATH .cache/fonts``.

Author: ```suggestion
Cached fonts are stored in the application output folder (TODO find a link in the doc) ``/ .cache/fonts``.
```

> Cached fonts are stored in ``TODO PATH .cache/fonts``.

Author: TODO...

actually you have to make the same changes than the dev guide

You may delete this directory to force the generation of all fonts in your application.

> You may also delete a single font in this directory so that this specific font is regenerated.

Author: hmm I am not sure... The cache info file is used as reference

Author: delete this line: this is not an option

> You may also delete a single font in this directory so that this specific font is regenerated.

Author: no
